resources/uco_design_document.md

```diff
@@ -71,24 +71,56 @@ To achieve its targeted objectives UCO has chosen to pursue an ontological appro
+In the domain graph, any individual instances of `UcoInherentCharacterizationThing` would be considered internal characteristic properties of the individual instances of `UcoObject` they are associated with.
+This means that any object properties in a UCO domain graph that do not have a range that is `UcoObject` or one of its subclasses MUST have a range that is `UcoInherentCharacterizationThing` or one of its subclasses.
+
+The common default pattern for specifying a relationship between domain concept class (`UcoObject`) instances and any characterizing non-domain concept class (`UcoInherentCharacterizationThing`) instances is through the definition of purpose-specific object properties for each such relationship. For example, in UCO, Windows PE binary files are represented using the `observable:WindowsPEBinaryFile` subclass of `UcoObject`, sections of such files are represented using the `observable:WindowsPESection` subclass of `UcoInherentCharacterizationThing` and the two are associated to each other using the purpose-specific object property `observable:sections`.
+
+The upside of the purpose-specific object property approach is clarity but there are limitations/downsides as well. These limitations/downsides as well as the chosen solution to address them (Facets) are further detailed in #5 below.
 
-Domain concept classes represent the things whereas facet classes represent the thing's characteristics. The disjointness between them follows from the fact that the thing can never be the same as its characteristics.
 
 All objects in UCO must specify a globally unique identifier (discussed in #4 below) and an assertion of the class type of the object.
 
-Instances of UcoObject subclasses (domain concept classes) are the granularity of discourse in the cyber domain and are thus objects/nodes in the UCO domain graph. Relationships between UcoObject subclasses (either in the form of OWL ObjectProperties or in the form of Relationship object subclasses of UcoObject as described below) are edges in the domain graph. Some relationships between UcoObject subclasses may require further characterization beyond simply expressing an association. These relationships are represented with the Relationship class which itself is a subclass of UcoObject and therefore a node itself in the domain graph. This is further discussed in #6 below.
+Instances of UcoObject subclasses (domain concept classes) are the granularity of discourse in the cyber domain and are thus objects/nodes in the UCO domain graph. Relationships between UcoObject subclasses (expressed as object properties) are edges in the domain graph. Some relationships between UcoObject subclasses may require further characterization beyond simply expressing an association. These relationships are represented with the Relationship class which itself is a subclass of UcoObject and therefore a node itself in the domain graph. This is further discussed in #6 below.
```

Comment:
The revised parenthetical was correct before the revision, and is wrong with the revision; see especially my remark above on the incongruity with `ObjectProperty`.  I will revert this change.

Reply:
Oh, the `diff` in the Files view on Github showed me a different edit to the parenthetical.

I'm still not quite sure I'm happy with where the parenthetical ended up.